### PR TITLE
Key management

### PR DESCRIPTION
**Pull request by isantop**
Adds the ability to add and remove signing keys, plus view signing key information, to Repoman.

Using new APIs from Repolib, signing keys can be added, viewed, and removed from repositories using Repoman.

New features will require the `popdev:gpg-info` staging branch with updated Repolib until that PR is merged. Testing without the branch should still allow Repoman to function correctly, albeit without the updated functionality.

Testing:
* Ensure no regressions from previous functionality
* Ensure that the "Signing Key Info" tab appears in the edit dialog for repositories which have a signing key added.
* Ensure that keys for existing repositories (like `popdev-master`) can be viewed and that information matches that provided by `apt-manage`
* Ensure that the "Add signing key" button appears on sources which do not have a key
* Ensure that keys can be added to existing repositories from URLs, Fingerprints, Paths on disk, and ASCII-Armor.
* Ensure that keys added to repositories can be removed from the source.
* For fingerprint key testing, use `1E5F8BBC5BEB10AE` which is the Pop OS PPA signing key fingerprint.
* Ensure that adding a key using a Fingerprint, URL, or ASCII-Armor add a new key file in `/etc/apt/keyrings` and that the file is removed when the key is removed from the source.
* For adding a URL, use `https://raw.githubusercontent.com/pop-os/pop/master/scripts/.iso.asc`. The text at that URL can also be used for testing ASCII-Armored keys
* For adding a path, use `/etc/apt/keyrings/popdev-archive-keyring.gpg`.
* Ensure that adding a key from a Path does **not** add a new file to `/etc/apt/keyrings` and that the key file is **not** removed when the key is removed from the source.
* Keys can be test-added on a single source (i.e. the `pop-os-release` source) then removed to test another method of adding.
* Ensure that if testing without the repolib `popdev:gpg-info` branch added, none of the changes appear and that Repoman continues to function correctly.

Soft-requires pop-os/repolib#60

### CODE REVIEW

**Review by n3m0-22 — CHANGES_REQUESTED** [review]
So far this is looking good, and testing is passing. I have found a few issues that should be addressed. 

Steps to reproduce:

Using `popdev-master` as an example

1. add a source
2. remove the key
3. add key using Fingerprint
4. add a typo to the Keyserver (e.g., .con instead of .com)

The `Add Signing Key` dialog box freezes and must be forced closed.

When using the URL method to add a key if the wrong URL is entered the source can no longer be opened for editing in Repoman. You can however remove the key with `apt-manage` to get it working again.

**Comment by isantop** [comment]
In my testing, 1c3e9f0 which I just pushed catches errors from both of the described situations (and probably many more besides) and fixes those issues. Thanks for pointing that out!

**Comment by isantop** [comment]
I added a couple of error messages to better handle if a user adds invalid ASCII-armored text as the signing key. There is also the possibility to remove the now-faulty key directly from the error message.

**Comment by isantop** [comment]
Screenshot of the faulty-key message: 

image

**Review by n3m0-22 — CHANGES_REQUESTED** [review]
Regression testing is passing, and the new error handling looks good!

My one reservation in terms of appearance is that I think it looks a bit strange to have a colon at the end of some lines as well as a period.
Screenshot from 2022-11-08 13-38-57

This colon looks particularly out of place when prompted to remove a bad ASCII-Armor key.
Screenshot from 2022-11-08 13-33-22

Functionally I see something that doesn't make sense to me. If you add a bad ASCII-Armor key you are not prompted about it until you try to edit it again, or by running `apt update`. It would be good if the error could be checked for prior to this point. Preferably when adding the key. If this is not an option then I would suggest removing the `Continue` button next to `Remove Key`. It leads back to Modify Source but without any options to remove or change the key. At that point the options are `Cancel` or `Save` and both lead to the same outcome. The bad key is preserved, and the removal process needs to start again.
Screenshot from 2022-11-08 13-45-35

**Comment by isantop** [comment]
> My one reservation in terms of appearance is that I think it looks a bit strange to have a colon at the end of some lines as well as a period.

I don't actually know why a colon was added to that dialog in the first place. I've removed the colon as of ff91abb

> Functionally I see something that doesn't make sense to me. If you add a bad ASCII-Armor key you are not prompted about it until you try to edit it again, or by running apt update. It would be good if the error could be checked for prior to this point. Preferably when adding the key. 

Loading the key before saving the source seemed to cause a lot of issues in my testing related to apparent race conditions getting the source saved, which is why saving the source with the bad key was the solution I decided on. Additionally, the below UI is still required as bad keys can be added in ways that don't involve Repoman, or even Repolib at all; they can be added by packages, by external tools, or even by manually downloading the key file and modifying the config files manually.

> If this is not an option then I would suggest removing the Continue button next to Remove Key. 

There's a couple of reasons that was added:
* If a key is actually valid but something in Repoman or Repolib is causing it to be detected improperly (key formatting or other issues possibly come to mind) it is important to have a non-destructive path forward without deleting the key. Since the rest of the modifications are also possible, there's no reason at that point to prevent any modifications to the source.
* The Remove option will require root privileges, and if the user does not have access to modify the source, there will be other error messages anyway, and we will fall through to the UI below. It's also preferable to provide an option to continue without requesting privileges in the first place.

> It leads back to Modify Source but without any options to remove or change the key. At that point the options are Cancel or Save and both lead to the same outcome. The bad key is preserved, and the removal process needs to start again.

Removing the Key details and "Add Key" button were deliberate decisions because with a bad key added, there are no details to view, and adding a new key would leave both the bad key and good key truncated together, which is not good. The most sane thing possible at this point, without removing the key, is ignoring the key management. 

As mentioned above, this UI is still required even without a Continue button because the user may click "Remove", then be unable to authenticate as a privileged user, and adding a new key

**Review by n3m0-22 — APPROVED** [review]
That all makes sense to me, and addresses all my issues. Looks good!

File: debian/changelog [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
+
+ -- Ian Santopietro <ian@system76.com>  Mon, 07 Nov 2022 10:36:30 -0700
+
+repoman (2.2.0) jammy; urgency=medium
 
   * Added support for renaming repositories
   * Added ability to change system repository mirrors
 
- -- Ian Santopietro <ian@system76.com>  Wed, 21 Oct 2020 10:57:09 -0600
+ -- Ian Santopietro <ian@system76.com>  Mon, 07 Nov 2022 10:36:07 -0700
```
Comment (13r0ck) [inline debian/changelog:12]: Why do both versions have the same release date?

Comment (isantop) [reply debian/changelog:12]: I discovered that the previous version was still listed as `UNRELEASED` and short of "cheating" the version to be released, marking it released using the correct tools also updates the date. I can revert that specific change if that would be preferred.

**Review by 13r0ck — APPROVED** [review]

**Pull request merged**